logModulesLoadedBeforeTrace should log a warning not an error, and should have a flag to disable the warning

The function [logModulesLoadedBeforeTrace](https://github.com/googleapis/cloud-trace-nodejs/blob/master/src/tracing.ts#L79-L106) will currently log an error, which is not appropriate for a warning to a developer. 

It should be changed to log a warning, so as to not trigger error reports when a service is restarted. Ideally the module should also have an option to disable this warning. 

I'm happy to make a PR to do this if considered a valid change.

@kjin Is there a branch that I should work off of? 

Master doesn't build:

```
cloud-trace-nodejs (master) λ npm run compile

> @google-cloud/trace-agent@4.0.1 compile /home/owg1/Projects/cloud-trace-nodejs
> npm run script get-plugin-types compile-auto compile-auto-strict


> @google-cloud/trace-agent@4.0.1 script /home/owg1/Projects/cloud-trace-nodejs
> ts-node -P ./scripts/tsconfig.json ./scripts "get-plugin-types" "compile-auto" "compile-auto-strict"

> Running step: get-plugin-types
> Running step: compile-auto
> Running: `node_modules/typescript/lib/tsc -p ./tsconfig.full.json --target es2017`
test/plugins/test-trace-mongoose-async-await.ts:34:29 - error TS2559: Type '{ f1: string; f2: boolean; f3: number; }' has no properties in common with type 'Partial<Document>'.

34     const data = new Simple(doc);
                               ~~~


Found 1 error.

Error: Process 18783 exited with code 2.
    at ChildProcess.childProcess.on (/home/owg1/Projects/cloud-trace-nodejs/scripts/utils.ts:41:14)
    at ChildProcess.emit (events.js:198:13)
    at ChildProcess.EventEmitter.emit (domain.js:448:20)
    at maybeClose (internal/child_process.js:982:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
```

--- a/src/tracing.ts
+++ b/src/tracing.ts
@@ -77,9 +77,9 @@ export class Tracing implements Component {
   }
 
   /**
-   * Logs an error message detailing the list of modules that were loaded before
-   * the Trace Agent. Loading these modules before the Trace Agent may prevent
-   * us from monkeypatching those modules for automatic tracing.
+   * Logs an warning message detailing the list of modules that were loaded
+   * before the Trace Agent. Loading these modules before the Trace Agent may
+   * prevent us from monkeypatching those modules for automatic tracing.
    * @param filesLoadedBeforeTrace The list of files that were loaded using
    * require() before the Stackdriver Trace Agent was required.
    */
@@ -97,7 +97,7 @@ export class Tracing implements Component {
       }
     }
     if (modulesLoadedBeforeTrace.length > 0) {
-      this.logger.error(
+      this.logger.warn(
         'StackdriverTracer#start: Tracing might not work as the following modules',
         'were loaded before the trace agent was initialized:',
         `[${modulesLoadedBeforeTrace.sort().join(', ')}]`

--- a/test/test-modules-loaded-before-agent.ts
+++ b/test/test-modules-loaded-before-agent.ts
@@ -44,8 +44,8 @@ describe('modules loaded before agent', () => {
     testTraceModule.start();
     assert.strictEqual(
       logger.getNumLogsWith(
-        'error',
-        /modules.*loaded.*before.*trace agent.*: .*shimmer/
+        'warn',
+        /StackdriverTracer#start.*modules.*loaded.*before.*trace agent.*: \[.*shimmer.*\]/
       ),
       1
     );

